src/core/theme/index.ts

```diff
@@ -74,6 +83,9 @@ export const suomifiTheme = ({
  * @param {function} baseStyles Function that will get components' props including tokens-prop and return CSS-styles
  */
 export const withSuomifiTheme = (
-  baseStyles: <K>(props: K & SuomifiThemeProp) => FlattenSimpleInterpolation,
-) => <T extends SuomifiThemeProp>({ tokens, ...passProps }: TokensProp & T) =>
-  baseStyles({ ...passProps, theme: suomifiTheme(tokens) });
+  baseStyles: <K>(props: K & TokensAndTheme) => FlattenSimpleInterpolation,
```

Comment:
This typing is always a pleasure to read! I would consider breaking it down for the sake of readability. Not as part of this PR though.